cesium/internal/domain/file_controller.go

```diff
@@ -126,7 +128,11 @@ func (fc *fileController) newWriter(ctx context.Context) (*controlledWriter, err
 	fc.writers.Lock()
 	fc.writers.open[nextKey] = w
 	fc.writers.Unlock()
-	return &w, err
+	if err != nil {
```

Comment:
What's the point of this check?

Reply:
thought it wasn't good to have `span.Error(nil)`. turns out that's fine.